blocks/library/gallery/index.js

```diff
@@ -114,14 +114,24 @@ registerBlockType( 'core/gallery', {
 				isMatch( files ) {
 					return files.length !== 1 && every( files, ( file ) => file.type.indexOf( 'image/' ) === 0 );
 				},
-				transform( files ) {
-					return Promise.all( files.map( ( file ) => createMediaFromFile( file ) ) )
-						.then( ( medias ) => createBlock( 'core/gallery', {
-							images: medias.map( media => ( {
-								id: media.id,
-								url: media.source_url,
-							} ) ),
-						} ) );
+				transform( files, onChange ) {
+					const block = createBlock( 'core/gallery', {
+						images: files.map( ( file ) => ( {
+							url: window.URL.createObjectURL( file ),
+						} ) ),
+					} );
+
+					Promise.all( files.map( ( file ) =>
```

Comment:
Minor: should we use Promise.all here and wait for all images to load or update each image after they load?
On the gallery block insertion, we update image by image. But I'm fine with merging it is as I think the code is simpler and then we can iterate and make behaviors identical / reuse code if possible.

Reply:
Good catch, I was wondering the same thing. This code is slightly simpler though :)